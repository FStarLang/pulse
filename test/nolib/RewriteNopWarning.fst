module RewriteNopWarning

#lang-pulse
open Pulse.Nolib

assume val foo : int -> slprop

(* The rewrite is a nop. *)
fn test1 (y : int)
  preserves foo 1
{
  rewrite each y as 1;
}

(* Same *)
fn test2 (y : int) (z : int)
  preserves foo 1 ** foo y
{
  rewrite each z as 42;
}

