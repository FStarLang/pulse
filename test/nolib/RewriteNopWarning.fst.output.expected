* Warning <unknown> at RewriteNopWarning.fst(11,1-12,21):
  - No rewrites performed.
  - Rewriting in:
      RewriteNopWarning.foo 1

* Warning <unknown> at RewriteNopWarning.fst(18,1-19,22):
  - No rewrites performed.
  - Rewriting in:
      RewriteNopWarning.foo y ** RewriteNopWarning.foo 1

